src/magnet.js

```diff
@@ -80,10 +80,8 @@ class Magnet {
      */
     this.plugins_ = [];
 
-    this.setupMiddlewares_();
-    this.setupApplicationSettings_();
-    this.registerWebpackConfig_();
     this.registerPlugins_();
+    this.registerWebpackConfig_();
```

Comment:
Why the order was changed here of `registerPlugins_` and `registerWebpackConfig_`?

Reply:
I was thinking about that. And that would be the only way to change the behavior of a plugin. I can revert it. But when I was refactoring, it seems to be the proper way so the user could overwrite stuff that would be introduced by a magnet plugin.

Reply:
NVM, doesn't matter, plugins are just called on start function. The order doesn't affect anything actually. I will revert. But there's no change in functionality.